src/pool.rs

```diff
@@ -132,37 +133,66 @@ pub struct ConnectionPool {
 
     /// Pool configuration.
     pub settings: PoolSettings,
+
+    /// AuthInfo
+    pub auth_hash: Option<String>,
 }
 
 impl ConnectionPool {
+    async fn auth_hash_has_changed(&mut self) -> bool {
```

Comment:
Same comment as above. I think auth passthrough should be configured at the pool level and this check won't be necessary then.

Reply:
See bellow